app/src/main/java/com/example/task1/SharedPrefManager.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.task1
+
+import android.content.Context
+import android.content.SharedPreferences
+
+class SharedPrefManager (private val context : Context){
+    private val sharedPreferences : SharedPreferences = context.getSharedPreferences("taskSharedPreferences",Context.MODE_PRIVATE)
+
+    fun getMovieSavedStatus(id : Int) : Int {
+            return sharedPreferences.getInt("$id",0)
+    }
+
+    fun setMovieSavedStatus(id : Int,value : Int){
+            val editor=sharedPreferences.edit()
+            editor.putInt("$id",value).apply()
+    }
+
+    fun removeMovieFromSharedPref(id : Int){
+        val editor=sharedPreferences.edit()
+        editor.remove("$id").apply()
+    }
+
+    fun getAllKeys() : MutableList<Int>{
+        val allKeysList : MutableList<Int> = mutableListOf()
+        for ( key in sharedPreferences.all.keys){
```

Comment:
you should save movies inside an array and then save the array in shared pref, think about case that you save different data in the sharedpref then how you can know  if the key is related to movie or not?